### ISSUE TITLE
How to approach drag-drop behavior?

### ISSUE THREAD

**Issue by Tokatak** [Real-Serious-Games/Unity-Weld#28]
One way or another it requires a returning argument while binding. Can't think of anything else.

**Comment by MorganMoon** [comment]
Drag and drop should be completely view related and view models should generally not care if something is drag and drop or not. Unity has some components that would help with creating something like this (EventTrigger). When I need to do fancy view stuff, I may create what I refer to as a "View Controller" that is generally just glue for special stuff the view needs to do. If you wanted to make a drag and drop view controller, unity has some interfaces for that, such as: https://docs.unity3d.com/ScriptReference/EventSystems.IDragHandler.html These interfaces are called from the event system component, you could then bind whatever it is you need from your view controller to a view model

**Comment by Tokatak** [comment]
Thank you for response but it looks like that i wasn't clear enough.
Dragging part and dropping is fine. The trick is to backtrack from view the required viewModel.

A simple example : 
`
public class ParentViewModel 
{
 ObservableList<ChildViewModel> _listA;
 ObservableList<ChildViewModel> _listB;
}
`
Both Lists are bound to some **containers** using collection binding and appropriate **items** .
**items**implement required I*smth*Drag , **containers** implement IDropHandler.
This way destination container ends with an gameobject (a view basically with unknown nature) that needs to be tracked and later on used to notify ParentViewModel  about required changes.
Both of this confuses me. 

edit:
Notifying part looks like could be done [this way](https://github.com/Real-Serious-Games/Unity-Weld/issues/12)

So to summarize:
Is there a way to track bound view model ? (from template perhaps?)
Or perhaps a more natural way of implementing drag and drop feature.

Thank you @MorganMoon  for helping, please share any thoughts on this.

**Comment by MorganMoon** [comment]
Hey, sorry about my misunderstanding, and sorry if I continue to misunderstand, but it looks like you want the "ParentViewModel" to be a dependency of "ChildViewModel", as is shown in the other issue (#12) that you linked. A super easy way to manage it as you said may be from the template (Or any implementation of IViewModelProvider) as you can call GetViewModel() to get the instance of the view model. However, I would recommend looking into Inversion of Control pattern/Dependency injection to track and manage and inject dependencies of your view models

**Comment by Tokatak** [comment]
IViewModelProvider is the thing i looked for. Just as you mentioned it was looking though. 
Was going to use Unity-Weld as a UI part of entity backend system. 
Not sure I get, how do you expect IOC to help in getting component, not like it is being created or smth. 
Could you clarify?

**Comment by MorganMoon** [comment]
Cool, glad I could help. Sorry if my wording was confusing, I meant you could use IOC instead of using the IViewModelProvider to get the instance of your dependent view model, and "ParentViewModel" could be injected into "ChildViewModel" as it seems in your case it would be a dependency of it. Or if you were wondering how to use IOC along with unity's component's I would recommend "Zenject", it's a great DI framework meant for Unity

**Comment by Tokatak** [comment]
=) Yep, Zenject is a wonderful thing, have some experience with it.
Oh, I get what you mean by injecting in models. Yes, in most likely should be this way , but will do with manual construction injection for now . 
Thank you for helping!
